main.cpp

```diff
@@ -121,385 +163,563 @@ struct CarWash
- */
 struct Laptop
 {
-	//number of letter keys
 	int numLetterKeys = 26;
-	//number of numeric keys
 	int numNumericKeys = 10;
-	//amount of binary information in hard disk
 	float binaryInfoInHardDisk = 499.96f;
-	//number of usb ports
 	int numUSBPorts = 2;
-	//amount of electricity needed for portable use
 	float electricityForPortableUse = 58.2f;
-	
-	struct User //first nested UDT
+
+	struct User
 	{
 		bool isUserMale = true;
 		int userAge = 37;
-		bool isUserAdmin = true;
+		bool isUserAdmin;
 		std::string userName = "Billy Crystal";
 		std::string password = "123456789";
+        int personalEmailTotal; // added member variable for nested 2nd implementation
+        int programmesOpen; // added member variable for nested 3rd implementation
 		
-		bool turnOnLaptop(bool laptopOff = true, bool userPressedPowerButton = true);
-		int sendEmails(int emailsPerDay = 10, bool personalEmails = true);
-		void openAProgram(int currentProgramAmount, bool isUserAdmin);
+		void turnOnLaptop(bool laptopOff, bool userPressedPowerButton);
+		auto sendEmails(bool personalEmails = true, int emailsPerDay = 10);
+		auto openAProgram(int currentProgramAmount, bool newProgrammeOpen);
 	};
 
-	//transfer information from usb device to hard disk
 	void transInfoFromUSBToHardDisk(float transferTime, float dataSize);
-    	//automatically adjust display brightness
 	void automaticallyAdjustBrightness(float timeOfDay, int displayBrightnessLevel);
-    	//provide update information to user
-	bool displayUpdateInfo(bool isUpdateAvailable, User user);
+	auto displayUpdateInfo(bool isUpdateAvailable, User user);
 
-	User whoUsingLaptop; //example of member variable whose type is a UDT (User)
+	User userAdmin;
 };
 
-/*
-Thing 2) MIDI Keyboard
-5 properties:
-    1) number of white keys
-    2) number of black keys
-    3) amount of user presets
-    4) number of velocity sensitive pads
-    5) number of encoders
-3 things it can do:
-    1) output MIDI information
-    2) display MIDI note value
-    3) consume electricity via USB port
- */
+void Laptop::User::turnOnLaptop(bool laptopOff, bool userPressedPowerButton)
+{
+   laptopOff = true;
+   userPressedPowerButton = true;
+}
+
+auto Laptop::User::sendEmails(bool personalEmails, int emailsPerDay)
+{
+ if(personalEmails == true)
+ {
+    emailsPerDay = 10;
+ }
+ else
+ {
+    emailsPerDay = 0;
+ }
+personalEmailTotal += emailsPerDay;
+}
+
+auto Laptop::User::openAProgram(int currentProgramAmount, bool newProgrammeOpen)
+{
+    currentProgramAmount = 3;
+    newProgrammeOpen = true;
+    programmesOpen += 1;
+}
+
+void Laptop::transInfoFromUSBToHardDisk(float transferTime, float dataSize = 200)
+{
+    transferTime = dataSize / 50;
+    binaryInfoInHardDisk += dataSize;
+}
+
+void Laptop::automaticallyAdjustBrightness(float timeOfDay, int displayBrightnessLevel)
+{
+    if(timeOfDay > 6.0f && timeOfDay < 18.0f)
+    {
+        displayBrightnessLevel = 80;
+    }
+    else
+        displayBrightnessLevel = 35;
+}
```

Comment:
If you're going to use `{ }`, use it for both `if` and `else` blocks.
- [ ] revise